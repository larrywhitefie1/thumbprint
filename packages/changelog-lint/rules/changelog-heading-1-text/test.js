const test = require('../../test-utils');
const rule = require('./');

const valid = ['# Changelog'];
const invalid = ['# Change Log', '# Change  Log', '# Changeee  Log'];

test({
    rule,
    valid,
    invalid,
});
