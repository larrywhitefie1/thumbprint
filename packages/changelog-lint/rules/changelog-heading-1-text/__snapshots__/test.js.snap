// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "# Change Log",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 13,
          "line": 1,
          "offset": 12,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Top level heading must say “Changelog” (“Change Log” found).",
      "ruleId": "changelog-heading-1-text",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "# Change  Log",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Top level heading must say “Changelog” (“Change  Log” found).",
      "ruleId": "changelog-heading-1-text",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #2 fails 1`] = `
Object {
  "input": "# Changeee  Log",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 16,
          "line": 1,
          "offset": 15,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Top level heading must say “Changelog” (“Changeee  Log” found).",
      "ruleId": "changelog-heading-1-text",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
